src/test/regress/sql/multi_transaction_2pc.sql

```diff
@@ -0,0 +1,36 @@
+--
+-- MULTI_TRANSACTION_2PC
+--
+-- Tests that check 2PC is used only for connections that perform a write
+SET citus.next_shard_id TO 1410000;
+SET citus.force_max_query_parallelization TO ON;
+CREATE SCHEMA multi_transaction_2pc;
+SET search_path = 'multi_transaction_2pc';
+SET citus.shard_replication_factor to 1;
+
+-- check that read-only participants skip prepare
+-- only two of the connections will perform a write (INSERT)
+CREATE TABLE test (a int);
+SELECT create_distributed_table('test', 'a');
+INSERT INTO test SELECT i FROM generate_series(0, 5)i;
+SET citus.log_remote_commands TO ON;
+BEGIN;
+INSERT INTO test VALUES (1);
```

Comment:
Instead of having settnig logging the remote commands on and off for this block, you could possibly use `set local citus.log_remote_commands to on;` instead the transaction block, the effects of which will end afer commit/rollback.